compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <vector>
+
+namespace
+{
+
+class PatternSimulator
+{
+public:
+  PatternSimulator(const luci::CircleTranspose *f_tr, const luci::CircleReshape *m_rs,
+                   const luci::CircleTranspose *b_tr)
+    : _front_transpose(f_tr), _mid_reshape(m_rs), _back_transpose(b_tr), _start_tag(0),
+      _can_remove_transpsoe(true)
+  {
+  }
+
+  void simulate();
+  bool canRemoveTransposes() const;
+  luci::CircleReshape *createAlternative(loco::Graph *graph);
+
+private:
+  void init_shape(const luci::CircleNode *in_tensor);
+  void simulate_transpose(const luci::CircleTranspose *transpose_node);
+  void simulate_reshape(const luci::CircleReshape *reshape_node);
```

Comment:
How about returning `bool`, which returns true only when analysis can proceed?

```suggestion
  bool simulate_transpose(const luci::CircleTranspose *transpose_node);
  bool simulate_reshape(const luci::CircleReshape *reshape_node);
```

Then it'll be possible to remove `_can_remove_transpose`.